service-search-nls/src/main/java/fi/nls/oskari/search/channel/OgcApiSearchChannel.java

```diff
@@ -0,0 +1,202 @@
+                item.setType(placetypeCodes.get(properties.getInt(localAdditionalInfoNames[0])).get(properties.getString("language")));
+                item.setLon(Double.parseDouble(coordinates.get(0).toString()));
+                item.setLat(Double.parseDouble(coordinates.get(1).toString()));
+                item.setZoomScale(properties.getDouble("scaleRelevance"));
+                item.setRegion(municipalityCodes.get(properties.getInt(localAdditionalInfoNames[1])).get(properties.getString("language")));
+                result.addItem(item);
+
+            }
+
+        } catch (IOException ex) {
+            throw new RuntimeException("Error searching", ex);
+        } catch (JSONException e) {
+            log.warn("Json error during building and/or making of Search results", e);
+        }
+        return result;
+    }
+
+    /**
+     * Chooses when ever to use local or url based resource to make code hashmaps.
+     *
+     * @param thing
+     * @return
+     */
+    private HashMap<Integer, HashMap<String, String>> buildAdditionalInfo(String thing) {
+        HashMap<Integer, HashMap<String, String>> results;
+        if (USE_LOCAL_ADDITIONAL_INFO) {
+            results = buildAdditionalInfoFromLocals(thing);
+        } else {
+            results = buildAdditionalInfoFromURL(additionalPlaceTypeInfoBaseUrl + thing + "/items");
+        }
+        return results;
+    }
+
+    /**
+     * Takes url suffix in and constructs string represantation of that file.
+     *
+     * @param infoURL
+     * @return HashMap that holds id to hashmap of languages.
+     */
+    private HashMap<Integer, HashMap<String, String>> buildAdditionalInfoFromURL(String infoURL) {
+        HashMap<Integer, HashMap<String, String>> results = new HashMap<>();
+        for (int i = 0; i < languageArray.length; i++) {
+            try {
+                final String responseData = IOHelper.getURL(infoURL + "?lang=" + languageArray[i]);
+                buildLanguageInfoMap(results, responseData);
+
+            } catch (IOException ex) {
+                throw new RuntimeException("Error getting additional information", ex);
+            }
+        }
+        return results;
+    }
+
+    /**
+     * Takes file suffix in and constructs string represantation of that file.
+     *
+     * @param fileSuffix
+     * @return HashMap that holds id to hashmap of languages.
+     */
+    private HashMap<Integer, HashMap<String, String>> buildAdditionalInfoFromLocals(String fileSuffix) {
+        HashMap<Integer, HashMap<String, String>> results = new HashMap<>();
+        for (int i = 0; i < languageArray.length; i++) {
+            InputStream inputStreamFromFile = OgcApiSearchChannel.class.getResourceAsStream(ID + "_" + fileSuffix.toLowerCase() + "_" + languageArray[i] + ".json");
+            Scanner s = new Scanner(inputStreamFromFile, "UTF-8").useDelimiter("\\A");
+            String fileAsString = s.hasNext() ? s.next() : "";
```

Comment:
Is there some reason for using Scanner? There's a function in IOHelper that makes this easier to read:
```
String fileAsString = IOHelper.readString(getClass().getResourceAsStream((ID + "_" + fileSuffix.toLowerCase() + "_" + languageArray[i] + ".json"));